feat-add-annlite-filter (#331)
jina-ai/docarray: docs/advanced/document-store/annlite.md

| `max_connection`  | The number of bi-directional links created for every new element during construction. | `None`, defaults to the default value in the AnnLite package* |

*You can check the default values in [the AnnLite source code](https://github.com/jina-ai/annlite/blob/main/annlite/core/index/hnsw/index.py)

## Search with filter

Search with `.find` can be restricted by user-defined filters. Such filters that can be constructed using the following operators:

| Name              | Description            | Equivalent Python operator |
|-------------------|------------------------|----------------------------|
| `$gte`            | Greater or equal to    | `>=`                       |
| `$gt`             | Greater than           | `>`                        |
| `$lte`            | Less or equal to       | `<=`                       |
| `$lt`             | Less than )            | `<`                        |
| `$eq`             | Equal to               | `==`                       |
| `$neq`            | Not equal to           | `!=`                       |

> filters can be constructing following the guidelines provided in [the AnnLite source repository](https://github.com/jina-ai/annlite).

Reviewer: ```suggestion
Filters can be constructed following the guidelines provided in [the AnnLite source repository](https://github.com/jina-ai/annlite).
```

### Example of `.find` with a filter

Consider Documents with embeddings `[0,0,0]` up to ` [9,9,9]` where the document with embedding `[i,i,i]`
has as tag `price` with value `i`. We can create such example with the following code:

```python
from docarray import Document, DocumentArray
import numpy as np

n_dim = 3
metric = 'Euclidean'

da = DocumentArray(
    storage='annlite',
    config={'n_dim': n_dim, 'columns': [('price', 'float')], 'metric': metric},
)

with da:
    da.extend(
        [
            Document(id=f'r{i}', embedding=i * np.ones(n_dim), tags={'price': i})
            for i in range(10)
        ]
    )
```

Consider we want the nearest vectors to the embedding `[8. 8. 8.]`, with the restriction that

> prices must follow a filter. As an example, let us consider that retrieved documents must have `price` value lower

Reviewer: ```suggestion
prices must follow a filter. As an example, le's consider that retrieved documents must have `price` value lower
```

or equal than `max_price`. We can encode this information in annlite using `filter = {'price': {'$lte': max_price}}`.

Then the search with the proposed filter can implemented and used with the following code:

```python
max_price = 7
n_limit = 4

> np_query = np.ones(n_dim) * 8

Reviewer: ```suggestion
np_query = np.array([8, 8, 8])
```
maybe more readable like this

print(f'\nQuery vector: \t{np_query}')

filter = {'price': {'$lte': max_price}}
results = da.find(np_query, filter=filter, limit=n_limit)
print('\nEmbeddings Nearest Neighbours with "price" at most 7:\n')
for embedding, price in zip(results.embeddings, results[:, 'tags__price']):
    print(f'\tembedding={embedding},\t price={price}')
```

> This would print

Reviewer: ```suggestion
This would print:
```
